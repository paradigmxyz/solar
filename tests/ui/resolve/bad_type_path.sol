// Testing `ResolverError`.

struct S {
    uint x;
}

struct S2 {
    uint x;
}
struct S2 { //~ ERROR: already declared
    uint x;
}

function f() {}
function f(uint x) {}

contract C {
    S s1;
    S.x s2; //~ ERROR: `S` is a struct, which cannot be indexed in type paths

    S2 s3;
    S2.x s4; //~ ERROR: `S2` is a struct, which cannot be indexed in type paths

    f f1; //~ ERROR: symbol `f` resolved to multiple declarations
    f.x f2; //~ ERROR: symbol `f` resolved to multiple declarations
}
