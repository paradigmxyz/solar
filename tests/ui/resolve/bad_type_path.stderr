error: identifier `S2` already declared
  --> ROOT/tests/ui/resolve/bad_type_path.sol:LL:CC
   |
LL | struct S2 {
   |        -- note: previous declaration declared here
LL |     uint x;
LL | }
LL | struct S2 {
   |        ^^
   |

error: `S` is a struct, which cannot be indexed in type paths
  --> ROOT/tests/ui/resolve/bad_type_path.sol:LL:CC
   |
LL |     S.x s2;
   |     ^
   |

error: `S2` is a struct, which cannot be indexed in type paths
  --> ROOT/tests/ui/resolve/bad_type_path.sol:LL:CC
   |
LL |     S2.x s4;
   |     ^^
   |

error: symbol `f` resolved to multiple declarations
  --> ROOT/tests/ui/resolve/bad_type_path.sol:LL:CC
   |
LL |     f f1;
   |     ^
   |

error: symbol `f` resolved to multiple declarations
  --> ROOT/tests/ui/resolve/bad_type_path.sol:LL:CC
   |
LL |     f.x f2;
   |     ^
   |

error: aborting due to 5 previous errors

